src/app/core/auth/auth.service.ts

```diff
@@ -75,20 +82,32 @@ export class AuthService {
       });
   }
 
-  updateLoginDisplay(): void {
+  private updateLoginDisplay(): void {
     const isLoggedIn = this.msalService.instance.getAllAccounts().length > 0;
+    this._loginDisplay.next(isLoggedIn);
 
     if (isLoggedIn) {
       const account = this.msalService.instance.getActiveAccount();
       if (account) {
-        this.userId.set(account.localAccountId);
-        this._activeAccountReady.next(true);
+        const id = account.localAccountId;
+        this.appUserService.getUser(id).subscribe(user => {
+          const logged: AppUser = {
```

Comment:
Zmieniłbym AppUser na interface o nazwie LoggedUser żeby odpowiadał temu co to rzeczywiście przechowuje
+ po zmianie z BehaviorSubject na signal trzeba tutaj zrobić this.currentUser.set()